src/Catalyst.Core.Modules.Dfs/UnixFs/ChunkedStream.cs

```diff
@@ -0,0 +1,176 @@
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Core.Lib.Dag;
+using Lib.P2P;
+using ProtoBuf;
+
+namespace Catalyst.Core.Modules.Dfs.UnixFileSystem
+{
+    /// <summary>
+    ///   Provides read-only access to a chunked file.
+    /// </summary>
+    /// <remarks>
+    ///   Internal class to support <see cref="UnixFs"/>.
+    /// </remarks>
+    public class ChunkedStream : Stream
+    {
+        class BlockInfo
+        {
+            public Cid Id;
+            public long Position;
+        }
+
+        List<BlockInfo> blocks = new List<BlockInfo>();
+        long fileSize;
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="ChunkedStream"/> class with
+        ///   the specified <see cref="IBlockApi"/> and <see cref="DagNode"/>.
+        /// </summary>
+        /// <param name="blockService"></param>
+        /// <param name="keyChain"></param>
+        /// <param name="dag"></param>
+        public ChunkedStream(IBlockApi blockService, IKeyApi keyChain, IDagNode dag)
+        {
+            BlockService = blockService;
+            KeyChain = keyChain;
+            var links = dag.Links.ToArray();
+            var dm = Serializer.Deserialize<DataMessage>(dag.DataStream);
+            fileSize = (long) dm.FileSize;
+            ulong position = 0;
+            for (int i = 0; i < dm.BlockSizes.Length; ++i)
+            {
+                blocks.Add(new BlockInfo
+                {
+                    Id = links[i].Id,
+                    Position = (long) position
+                });
+                position += dm.BlockSizes[i];
+            }
+        }
+
+        IBlockApi BlockService { get; set; }
+        IKeyApi KeyChain { get; set; }
+
+        /// <inheritdoc />
+        public override long Length => fileSize;
+
+        /// <inheritdoc />
+        public override void SetLength(long value) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override bool CanRead => true;
+
+        /// <inheritdoc />
+        public override bool CanSeek => true;
+
+        /// <inheritdoc />
+        public override bool CanWrite => false;
+
+        /// <inheritdoc />
+        public override void Flush() { }
+
+        /// <inheritdoc />
+        public override long Position { get; set; }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            switch (origin)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)